vulture/core.py

```diff
@@ -163,6 +168,9 @@ def get_set(typ):
         self.used_names = get_set('name')
 
         self.ignore_names = ignore_names or []
+        self.ignore_decorators = ignore_decorators or []
+        self.ignore_decorators = [
+            decorator.lstrip('@') for decorator in self.ignore_decorators]
```

Comment:
With our current design ``--ignore-decorators "*@app.route"`` won't ignore the decorated functions even though it should. I doubt that anybody will ever use that, but it shows that maybe our design should be simplified. Let's change it so that get_decorator_name prepends an @ to the returned names and let's not strip leading @'s. Then "*@app.route", "@*app.route" and "*app.route" all work.

Reply:
Hmm, nice catch - Let me do the required. :-)

Thanks!

Reply:
> Then "*@app.route", "@*app.route" and "*app.route" all work.

But, dong this also breaks `app.route` since `fnmatchcase('app.route', '@app.route') != True`. Maybe rather than stripping `@`, we can add one if missing. What say?

Reply:
I ended up doing `'@' + decorator.lstrip('@') for decorator in ignore_decorators`. Please let me know if you see something wrong with this approach. :-)

Reply:
Oops, if the initial character of the input is a wildcard (e.g., *, ?) - this would fail. :-(

Reply:
ahh, yeah let's do that - I like the `@` syntax anyways - more clear, less chances of error.